src/GraphQL/Resolver.hs

```diff
@@ -105,11 +113,11 @@ ok :: GValue.Value -> Result GValue.Value
 ok = pure
 
 
--- TODO instead of SelectionSet we want something like
--- NormalizedSelectionSet which has query fragments etc. resolved.
+-- TODO: It's not crystal clear that "SelectionSet" should be the entry point.
```

Comment:
I think for `HasGraph` it's the correct entry point because we need to be able to nest, and there are no nested `Operation`s other than the top level one.

But yea: I think we need another top-level entry point as well to build the full resolver.